src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Protocol/SemanticTokensResponse.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Protocol
+{
+    internal class SemanticTokensResponse
+    {
+        public string ResultId { get; set; } = null;
+
+        public ICollection<uint> Data { get; set; } = new uint[0];
```

Comment:
And again here. Why not uint[], IList<uint>, IEnumerable<uint> or any number of more common/idiomatic interfaces

Reply:
Some of my thoughts process behind this.
1) If i knew how many tokens i am going to skip or how many multiline tokens i am going to break down then i would have created a uint[] array in EncodeTokens() method and write it using a pointer index.
2) But i need to run one more loop to compute this size so i had to use a dynamic list (LinkedList/List). I went with LinkedList for faster appends towards the end as each token is mapped to 5 tokens. We just serialize this uint collection in the end. 
3) I could convert the LinkedList or List (whichever dynamic list i use) to an uint[] array and return type of EncodeTokens in this case would be uint[]. This array would then be serialized to json array by the JsonSerializer. The question i asked myself was do i need to convert the dynamic list i used before to an array and then serialize that array. I could just directly serialize the dynamic list i used to build tokens. No need to do take an extra step of converting to array if in the end i am just going to iterate over it from left to right and serialize it.  
4) My bad on this that i didn't spend time thinking about whether to return dynamic list as ICollection or IEnumerable. IEnumerable makes sense for what i described above in the first 3 points. I can use that instead of ICollection.

Reply:
I changed this to IEnumerable. Let me know if you have any other feedback around this

Reply:
Resolving this. Let me know if you have any concerns about this